tests/bluetooth/host/keys/bt_keys_get_addr/src/main.c

```diff
@@ -134,36 +158,31 @@ void test_get_key_by_matched_id_and_address(void)
 	}
 }
 
-/* Setup test variables */
-static void unit_test_setup(void)
+static void fff_reset_rule_before(const struct ztest_unit_test *test, void *fixture)
 {
-	clear_key_pool();
-}
 
-void test_main(void)
-{
-
-	/*
-	 * - Checks the initial startup state of the keys pool to be empty
-	 * - Test bt_keys_get_addr() behaviour when trying to get a key
-	 *   reference for non-existing (ID, Address) pair which should result in adding the
-	 *   non-existing pair to the keys pool list
-	 */
-	ztest_test_suite(bt_keys_get_addr_get_non_existing_keys,
-			 ztest_unit_test(test_list_is_empty_at_startup),
-			 ztest_unit_test(test_populate_key_pool),
-			 ztest_unit_test(test_no_equal_references));
+	/* Skip tests if not all startup suite hasn't been executed */
+	if (strcmp(test->test_suite_name, "bt_keys_get_addr_startup")) {
+		zassume_true(all_startup_checks_executed == true, NULL);
+	}
 
-	ztest_run_test_suite(bt_keys_get_addr_get_non_existing_keys);
+	ASSERT_FFF_FAKES_LIST(RESET_FAKE);
+	CONN_FFF_FAKES_LIST(RESET_FAKE);
+	HCI_CORE_FFF_FAKES_LIST(RESET_FAKE);
+}
 
-	/* Checks when (ID, Address) pairs exist in keys pool */
-	ztest_test_suite(bt_keys_get_addr_get_existing_keys,
-				 ztest_unit_test_setup(test_populate_key_pool, unit_test_setup),
-				 ztest_unit_test(test_get_key_by_matched_id_and_address));
+ZTEST_RULE(fff_reset_rule, fff_reset_rule_before, NULL);
 
-	ztest_run_test_suite(bt_keys_get_addr_get_existing_keys);
+void test_main(void)
+{
+	/* Only startup suite will run. */
+	all_startup_checks_executed = false;
```

Comment:
IMO, 
```c
zassert_true(check_key_pool_is_empty(),
		     "List isn't empty, make sure to run this test just after a fresh start");			     "List isn't empty, make sure to run this test just after a fresh start");
```
Should be part of the test case setup (`every_test_setup`), right after `clear_key_pool`. or even _in_ `clear_key_pool`.

Reply:
To make it simple, let's agree on not using `zassert_` checks in sources outside the project source files.
That leaves us with one choice to make that check part of the test cases setup.

Reply:
yep I was going to suggest that as an alternative.